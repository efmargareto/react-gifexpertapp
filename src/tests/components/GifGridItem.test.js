import '@testing-library/jest-dom'
import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe( 'Testeando Componente <GifGridItem />', () => {
    
    const title = 'Baloncesto'
    const url = 'https://localhost.com/img/baloncesto.jpg'
    const wrapper = shallow( <GifGridItem title={ title } url= { url } /> )

    test('Debe de mostrar el componente correctamente', () => {

        expect( wrapper ).toMatchSnapshot();
    })

    test('Debe de tener un parrafo con el titulo', () => {
        
        const p = wrapper.find('p')
        expect( p.text().trim() ).toBe( title )
    })

    test('Debe de tener la imagen igual al url y alt de los prop', () => {

        const img = wrapper.find('img').props()
        const { src, alt } = img

        expect( src ).toBe( url )
        expect( alt ).toBe( title )
    })

    test('Debet ener las clases de la animacion, animate__fadeInUp', () => {

        const div = wrapper.find('div').props()
        const { className } = div

        expect( className ).toContain('animate__fadeInUp')
    })
    
}) 