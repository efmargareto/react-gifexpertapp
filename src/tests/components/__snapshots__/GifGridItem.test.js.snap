// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testeando Componente <GifGridItem /> Debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeInUp"
>
  <img
    alt="Baloncesto"
    src="https://localhost.com/img/baloncesto.jpg"
  />
  <p>
     
    Baloncesto
     
  </p>
</div>
`;
